File: pset/js/week01_cash.js
// Variable
let amount = 0;

// Functions
function getChange() {
    do {
        amount = prompt("What's the total change?");
    } while (amount <= 0 || isNaN(amount));
}

function change(total) {
    // Round up the total amount to prevent error
    total = Math.round(amount * 100);
    // Create and count all the coins
    let quarters = Math.floor(total / 25);
    let dimes = Math.floor((total % 25) / 10);
    let nickles = Math.floor(((total % 25) % 10) / 5);
    let pennies = Math.floor(((total % 25) % 10) % 5);
    // Return answer
    return console.log(`Calculating change...

  •  0.25¢: ${quarters}
  •  0.10¢: ${dimes}
  •  0.05¢: ${nickles}
  •  0.01¢: ${pennies}
=========================
Total coins: ${quarters + dimes + nickles + pennies}`)
}

// Run it all
getChange();
change(amount);

File: pset/c/week01_cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    // prompt change until valid number
    float changeF;
    do
    {
        changeF = get_float("Change: ");        
    } 
    while (changeF <= 0);
    // round change
    int change = round(changeF * 100);
    // calculate coins
    int quarters = change / 25;
    int dimes = (change % 25) / 10;
    int nickles = ((change % 25) % 10) / 5;
    int pennies = (((change % 25) % 10) % 5);
    // print result
    printf("%i\n", quarters + dimes + nickles + pennies);
}